Kernel/EqHelper.hpp

```diff
@@ -49,6 +49,11 @@ class EqHelper
 
   //WARNING, this function cannot be used when @param t is a sort.
   static Term* replace(Term* t, TermList what, TermList by);
+  static TermList replace(TermList t, TermList what, TermList by) {
+    return t == what ? by 
+         : t.isVar() ? t
+         : TermList(replace(t.term(), what, by));
+  }
```

Comment:
Nice. I've needed the exact same function on `dedukti`, will be nice to have this in master.